File: hw/chicaiza/u3/HW16Unitest/taxesjs/Tax.js
// Tax.js
// Conversión de Tax.java a JavaScript

class Tax {
    computeTax(basePrice, percentage, charged) {
        if (!charged) {
            return 0.0;
        } else {
            return basePrice * percentage / 100;
        }
    }

    computeIce(basePrice) {
        // Aquí iría la lógica para calcular el ICE
        return 0.0;
    }
}

module.exports = Tax;

File: ws/suarez/U3/WS28UnitTest/WS28UnitTest/Taxes/src/main/java/ec/edu/espe/controller/Tax.java
package ec.edu.espe.controller;

/**
 *
 * @author BERNARDO SUAREZ
 */
public class Tax {

    public float computeTax(float basePrice, float percentage, boolean charged) {
        float ivaValue = 0.0F;

        if (!charged) {
            return 0.0F;
        } else {
            ivaValue = basePrice * (percentage / 100);
        }
        return ivaValue;
    }
    
    public float computeIce(float basePrice){
        //code to compute ICE
        return 0.0F;
    }
    
}
